src/main/java/uk/gov/hmcts/reform/prl/services/caseaccess/AssignCaseAccessService.java

```diff
@@ -33,13 +34,15 @@ public void assignCaseAccess(String caseId, String authorisation) {
 
             log.info("CaseId: {} of type {} assigning case access to user {}", caseId, CASE_TYPE, userId);
 
-            String serviceToken = authTokenGenerator.generate();
-            assignCaseAccessClient.assignCaseAccess(
-                authorisation,
-                serviceToken,
-                true,
-                buildAssignCaseAccessRequest(caseId, userId, CASE_TYPE)
-            );
+            if (userDetails.getRoles() != null && !userDetails.getRoles().contains(COURT_ADMIN_ROLE)) {
```

Comment:
line 37, try this - 
UserDetails userDetails = userService.getUserDetails(authorisation);
List<String> roles = userDetails.getRoles();
boolean isCourtStaff = roles.stream().anyMatch(ROLES::contains);

Reply:
Added as specified testing the logic